### ISSUE TITLE
Add support for Github specified roles

### ISSUE THREAD

**Issue by RCheesley** [gitconsensus/GithubOrganizer#8]
Hi there!

When assigning teams to repos there are a [specified roles](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-permission-levels-for-an-organization) that can be used:

![screenshot-github com-2021 04 24-16_16_55](https://user-images.githubusercontent.com/2930593/115963740-9f93e500-a518-11eb-906d-2e059cadc537.png)

At the moment it seems that we only have push, pull and admin available which isn't really granular enough for large open source projects (unless there is some other setting that I am missing!)

My suggestion would be to align with the the roles that are available to have:

- Read
- Triage
- Write
- Maintain
- Admin

I found the roles being used [here](https://github.com/gitconsensus/GithubOrganizer/blob/820475dff2a97b6e190eb10b591140b2fdb61131/githuborganizer/tasks/github.py#L145) but I am not sure how to update them to reflect these roles (I do not have a background in development, sorry!)

**Comment by tedivm** [comment]
I've love to do this but unfortunately the GitHub API only exposes Read, Write, and Admin as the permissions that can be set.

https://docs.github.com/en/rest/reference/teams#add-or-update-team-project-permissions

**Comment by RCheesley** [comment]
Hmm, that link relates to adding projects to teams:

> Adds an organization project to a team

However here: https://docs.github.com/en/rest/reference/teams#check-team-permissions-for-a-repository it makes reference to all the roles that I am referring to. 

Here we are talking about assigning teams to repos: https://organizer.gitconsensus.com/#/?id=assign-teams-and-permissions 

So I think that we should be using the commands that are referenced here: https://docs.github.com/en/rest/reference/teams#add-or-update-team-repository-permissions rather than the ones in the projects?

**Comment by RCheesley** [comment]
@tedivm just wondered if you might have a chance to take a look at this. 

I think it's just a case of the wrong permissions being used (it is using the permissions for assigning teams to projects, rather than the permissions for assigning teams to repos).  

They are two different sets of permissions so at the moment it doesn't seem to be usable, because there isn't the triage permission which is a pretty important one.

**Comment by tedivm** [comment]
Yup you're right, the endpoint [here](https://docs.github.com/en/rest/reference/teams#add-or-update-team-repository-permissions) does include those other options. I should be able to add this functionality in.
